pepper-apis/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -525,9 +552,9 @@ private static void respondToGAEStartHook(String envPort) {
                 LOG.warn("Wait interrupted", e);
             }
         }
-
-        Spark.stop();
-        Spark.awaitStop();
+        // Commenting out while we keep heapdump functionality
+        //        Spark.stop();
+        //        Spark.awaitStop();
```

Comment:
Instead of opening an HTTP endpoint, how about we add a `HEAP_DUMP` task instead? It seems to fit as an "operational task". Security-wise, only folks who has access to our GCP project can publish to the pubsub topic. You should be able to easily add it to `HousekeepingTaskReceiver`. See https://broadinstitute.atlassian.net/wiki/spaces/DDP/pages/1261797381/Housekeeping+Operational+Tasks.

Reply:
Production Housekeeping does not have a SparkJava HTTP server running, while it does have a pubsub subscriber for these tasks. So I'll recommend going this route to get a better reflection of memory usage.

Reply:
I agree using pub sub would be more in the spirit of Housekeeping. And the more I think about it, it is probably worth the additional effort. @zyme @yufengwng would you guys agree that keeping the on-demand heap dumps as permanent part of system is worthwhile? I had been thinking of this as tactical/do it and forget it capability, but I now believe we should keep this available for the next time too.

Reply:
3e73c4b34b38a4da66fff83a4471a559c71588b5

Reply:
Having this as a permanent option seems great!